core/spec/lib/spree/core/testing_support/factories/order_factory_spec.rb

```diff
@@ -122,6 +122,26 @@
     end
   end
 end
+
+RSpec.shared_examples 'supplied completed_at is respected' do
+  context 'when passed a completed_at timestamp' do
+    let(:completed_at) { 2.days.ago }
+    let(:order) { create(factory, completed_at: completed_at) }
+
+    it 'respects the timestamp' do
+      expect(order.completed_at).to be_within(5.seconds).of(completed_at)
+    end
+  end
+
+  context 'when no completed_at timestamp is passed' do
+    let(:order) { create(factory) }
+
+    it 'defaults to the current time' do
+      expect(order.completed_at).to be_within(5.seconds).of(Time.current)
```

Comment:
`1.second` should be plenty of leeway.

Reply:
As mentioned here, I actually ran into an issue with the 1s time when posting the PR. The test ran at the rollover to a new second. Clock time at the start of the test was `xx:yy:01.995`, mysql stored `xx:yy:01.000` and the comparison tested against `xx:yy:02.016` (or something similar to that). Happy to move it back down to 2s which should account for this, if desired.

Perhaps a better test than this would be to record the time before and after the factory runs and test that the completed_at is between those two values?

Reply:
Actually the range idea still doesn't work because of the truncation.

I'll just update it to be 2 seconds instead.

Reply:
```suggestion
      expect(order.completed_at).to be_within(2.seconds).of(Time.current)
```